src/common.ts

```diff
@@ -0,0 +1,96 @@
+import { ScannerViolation, ScannerViolationType } from "./sfdxCli";
```

Comment:
What's the purpose of segregating this logic into `common`? We already introduced an abstract class with `reporter.types` which is where I would imagine all shared type and function definitions to be defined.

Do you have a class diagram that can show these how these classes interact and relate to each other? I am having a difficult time understanding the separation of concerns in this proposed change

Reply:
This file was slightly a remnant before the TS port, I've cleaned this up, thanks for the feedback.

I've moved everything that is only used by the reporters to `reporter.types.ts` and left everything else as is.

Common is just for any functionality that might be used across the different parts of the plugin, they could be stored in `index.ts` but I'd rather avoid circular dependencies

Dependency tree:
```mermaid
flowchart TD
    A[common.ts]
    B[annoations-reporter.ts] --> A
    C[comments-reporter.ts] --> A
    D[index.ts] --> A
    D[index.ts] --> B
    D[index.ts] --> C
```

Reply:
This graph is _sweet_ - I was unaware of mermaid until now. Thanks for sharing!